File: Simbean/python/baek/b1260 DFSBFS.py
import sys
#https://www.acmicpc.net/problem/1260
def dfs(point,visit):
    for i in v[point-1]:
        if i in visit:
            pass
        else:
            visit.append(i)
            dfs(i,visit)
    return visit
def bfs(s,e,visit):
    #print(v)
    while True:
        #print(visit)
        for i in range(s,e):
            for j in v[visit[i]-1]:
                if j in visit:
                    pass
                else:
                    visit.append(j)
        if e == len(visit):
            break
        else:
            s = e
            e = len(visit)
    return visit
def init():
    global v,o
    v = []
    o = 0
    visit = [N[2]]
    for i in range(1000):
        v.append([])
    for i in range(N[1]):
        M = list(map(int, sys.stdin.readline().split()))
        v[M[0]-1].append(M[1])
        v[M[1]-1].append(M[0])
    for i in v:
        i.sort()
    visit = dfs(N[2],visit)
    for i in visit:
        print(i, end=' ')
    visit = [N[2]]
    visit= bfs(0,len(visit),visit)
    print()
    for i in visit:
        print(i, end=' ')
if __name__ == "__main__":
    d = []
    b = []
    N = list(map(int,sys.stdin.readline().split()))
    init()

File: Simbean/c++/baek/b1260 dfsbfs.cpp
#include <string>
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;
int i;
vector<vector<int>> v;
void print(vector<vector<int>> a) {
	int j;
	for (i = 0; i < a.size();i++) {
		for (j = 0; j < a[i].size(); j++) {
			cout << a[i][j];
		}
		cout << endl;
	}
}

void print(vector<int> a) {
	int j;
	for (i = 0; i < a.size();i++) {
			cout << a[i] << ' ';
		}
	cout << endl;
}
vector<int> dfs(int a, vector<int> visit) {
	int point;
	vector<int>::iterator it;
	for (it = v[a-1].begin();it < v[a-1].end();it++) {
		if (find(visit.begin(), visit.end(), *it) == visit.end()) {
			visit.push_back(*it);
			visit = dfs(*it, visit);
		}
	}
	return visit;
}
vector<int> bfs(int s, int e, vector<int>visit) {
	vector<int>::iterator it;
	while (1) {
		for (i = s; i < e; i++) {
			for (it = v[visit[i] - 1].begin(); it < v[visit[i] - 1].end();it++) {
				if (find(visit.begin(), visit.end(), *it) == visit.end()) {
					visit.push_back(*it);
				}
			}
		}
		if (e == visit.size()) {
			break;
		}
		else {
			s = e;
			e = visit.size();
		}
	}
	return visit;
}
int main() {
	int a, b, c;
	int d, e;
	cin >> a;
	cin >> b;
	v.assign(a, vector<int>(0, 0));
	cin >> c;
	vector<int> visit;
	visit.push_back(c);
	for (i = 0;i < b;i++) {
		cin >> d >> e;
		v[d-1].push_back(e);
		v[e-1].push_back(d);
	}
	for (i = 0; i < v.size(); i++) {
		sort(v[i].begin(), v[i].end());
	}
	//print(v);
	visit = dfs(c, visit);
	print(visit);
	visit.clear();
	visit.push_back(c);
	visit = bfs(0, visit.size(), visit);
	print(visit);
}